tests/test_views.py

```diff
@@ -0,0 +1,121 @@
+from django.contrib.auth import get_user_model
+from django.test import TestCase
+from django.urls import reverse
+
+from taxi.models import Manufacturer, Driver, Car
+
+INDEX_URL = reverse("taxi:index")
+MANUFACTURERS_URL = reverse("taxi:manufacturer-list")
+MANUFACTURERS_SEARCH_URL = reverse("taxi:manufacturer-list") + "?name=a"
+DRIVERS_URL = reverse("taxi:driver-list")
+DRIVERS_SEARCH_URL = reverse("taxi:driver-list") + "?username=a"
+CAR_URL = reverse("taxi:car-list")
+CAR_SEARCH_URL = reverse("taxi:car-list") + "?model=a"
+
+
+class PublicPagesTest(TestCase):
+    def test_index_login_required(self):
+        res = self.client.get(INDEX_URL)
+        self.assertNotEqual(res.status_code, 200)
+
+    def test_manufacturers_login_required(self):
+        res = self.client.get(MANUFACTURERS_URL)
+        self.assertNotEqual(res.status_code, 200)
+
+    def test_drivers_login_required(self):
+        res = self.client.get(DRIVERS_URL)
+        self.assertNotEqual(res.status_code, 200)
+
+    def test_cars_login_required(self):
+        res = self.client.get(CAR_URL)
+        self.assertNotEqual(res.status_code, 200)
+
+
+class PrivateManufacturerTest(TestCase):
+    def setUp(self) -> None:
+        self.driver = get_user_model().objects.create_user(
+            username="driver",
+            password="lkjhfdsa",
+            first_name="Driver",
+            last_name="Driverio",
+            license_number="OIU29032"
+        )
+        self.client.force_login(self.driver)
+
+    def test_search_manufacturer_by_name(self):

```

Comment:
test search in forms and here test correct retrieve of objects and proper template usage

Reply:
I don`t really get what is needed here, can you please elaborate?